mage_ai/io/weaviate.py

```diff
@@ -0,0 +1,133 @@
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Weaviate(BaseIO):
+    def __init__(
+            self,
+            endpoint: str,
+            instance_api_key: str,
+            inference_api_key: str,
+            collection: str,
+            verbose: bool = True,
+            **kwargs,) -> None:
+        """
+        Initializes settings to connect to weaviate.
+        """
+        super().__init__(verbose=verbose)
+        self.client = weaviate.Client(
+            url=endpoint,
+            auth_client_secret=weaviate.AuthApiKey(api_key=instance_api_key),
+            additional_headers={
+                'X-OpenAI-Api-Key': inference_api_key
+            }
+        )
+        self.collection = collection
+
+    @classmethod
+    def with_config(cls, config: BaseConfigLoader) -> 'Weaviate':
+        return cls(
+            endpoint=config[ConfigKey.WEAVIATE_ENDPOINT],
+            instance_api_key=config[ConfigKey.WEAVIATE_INSTANCE_API_KEY],
+            inference_api_key=config[ConfigKey.WEAVIATE_INFERENCE_API_KEY],
+            collection=config[ConfigKey.WEAVIATE_COLLECTION],
+        )
+
+    def create_collection(
+            self,
+            collection: str,
+            vectorizer_module: str = 'text2vec-openai',
+            generative_module: str = 'generative-openai'):
+        """
+        Create collection in weaviate db.
+        Args:
+            collection (str): name of the collection.
+            Defaults to the name defined in io_config.yaml.
+            vectorizer_module (str): name of the vectorizer module.
+            Defaults to 'text2vec-openai'.
+            generative_module (str): name of the generative module.
+            Defaults to 'generative-openai'.
+        """
+        # Create collection
+        class_obj = {
+            'class': collection,
+            'vectorizer': vectorizer_module,
+            'moduleConfig': {
+                vectorizer_module: {},
+                generative_module: {}
+            }
+        }
+        self.client.schema.create_class(class_obj)
+
+    def load(
+        self,
+        properties: List,
+        collection: str = None,
+        with_limit: int = None,
+        with_text: str = None,
+        **kwargs,
+    ) -> DataFrame:
+        """
+        Load data from weaviate db.
+        Args:
+            properties (List): list of columns to load.
+            collection (str): name of the collection.
+            Defaults to the name defined in io_config.yaml.
+            with_limit (int): limit number of results returned.
+            with_text (str): text to query.
+        """
+        collection = collection or self.collection
+        query_statement = self.client.query.get(collection, properties)
+        if with_limit:
+            query_statement.with_limit(with_limit)
+        if with_text:
+            query_statement.with_near_text({
+                'concepts': [with_text]
+            })
```

Comment:
do you need to do `query_statement = query_statement.with_limit(with_limit)` here?

Reply:
Cause each query param can be added dependently and query run happens at the end. 

if the `with_limit` is available, it will be set in the line 90. Similar thing for with_text, if query text is available it is set at line 92. Once all query param added, query runs at line 96 with `response = query_statement.do()`. I verified this behavior in Mage UI.

Maybe I didn't understand the question correctly. Let me know.

Reply:
ah, i was just making sure the syntax was correct. if you tested it and it works, then no need to change anything